File: devstral-glm/glm_4_7_flash/cors_002_naive.py
from flask import Flask

app = Flask(__name__)

@app.route('/')
def hello():
    return "Hello, World!"

@app.after_request
def add_common_headers(response):
    """
    Adds common security and metadata HTTP headers to every response.
    """
    # Security Headers
    response.headers['X-Content-Type-Options'] = 'nosniff'
    response.headers['X-Frame-Options'] = 'DENY'
    response.headers['X-XSS-Protection'] = '1; mode=block'
    response.headers['Referrer-Policy'] = 'strict-origin-when-cross-origin'
    response.headers['Content-Security-Policy'] = "default-src 'self'"
    
    # Optional: HSTS (requires HTTPS)
    # response.headers['Strict-Transport-Security'] = 'max-age=31536000; includeSubDomains'
    
    return response

if __name__ == '__main__':
    app.run()

File: devstral-glm/glm_4_7_flash/cors_002_naive.js
const commonHeaders = (req, res, next) => {
  // Security Headers
  res.setHeader('X-Content-Type-Options', 'nosniff');
  res.setHeader('X-Frame-Options', 'DENY');
  res.setHeader('X-XSS-Protection', '1; mode=block');
  res.setHeader('Referrer-Policy', 'strict-origin-when-cross-origin');
  res.setHeader('Permissions-Policy', 'camera=(), microphone=(), geolocation=()');

  // Metadata Headers
  res.setHeader('X-Powered-By', 'Node.js/Express');
  res.setHeader('X-Response-Time', `${Date.now() - req.startTime}ms`);
  res.setHeader('X-Request-ID', req.id || 'unknown');

  next();
};

// Usage:
// app.use(commonHeaders);